commons/ava_testsuite/byzantine_tests/unrequested_chit_spammer_network_tests.go

```diff
@@ -0,0 +1,111 @@
+	byzantineUsername = "byzantine_gecko"
+	byzantinePassword = "byzant1n3!"
+	stakerUsername = "staker_gecko"
+	stakerPassword = "stak3r1!"
+	normalNodeServiceId = 4
+	seedAmount               = int64(50000000000000)
+	stakeAmount              = int64(30000000000000)
+)
+// ================ Byzantine Test - Spamming Unrequested Chit Messages ===================================
+type StakingNetworkUnrequestedChitSpammerTest struct{
+	UnrequestedChitSpammerImageName string
+	NormalImageName                 string
+}
+func (test StakingNetworkUnrequestedChitSpammerTest) Run(network interface{}, context testsuite.TestContext) {
+	castedNetwork := network.(ava_networks.TestGeckoNetwork)
+	for i := 0; i < 4; i++ {
+		byzClient, err := castedNetwork.GetGeckoClient(i)
+		if err != nil {
+			context.Fatal(stacktrace.Propagate(err, "Failed to get byzantine client."))
+		}
+		highLevelByzClient := ava_networks.NewHighLevelGeckoClient(
+			byzClient,
+			byzantineUsername,
+			byzantinePassword)
+		err = highLevelByzClient.GetFundsAndStartValidating(seedAmount, stakeAmount)
+		if err != nil {
+			context.Fatal(stacktrace.Propagate(err,"Failed add client as a validator."))
+		}
+		currentStakers, err := byzClient.PChainApi().GetCurrentValidators(nil)
+		if err != nil {
+			context.Fatal(stacktrace.Propagate(err, "Could not get current stakers."))
+		}
+		logrus.Debugf("Number of current stakers: %d", len(currentStakers))
+	}
+	availabilityChecker, err := castedNetwork.AddService(normalNodeConfigId, normalNodeServiceId)
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Failed to add normal node with high quorum and sample to network."))
+	}
+	if err = availabilityChecker.WaitForStartup(); err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Failed to wait for startup of normal node."))
+	}
+	normalClient, err := castedNetwork.GetGeckoClient(normalNodeServiceId)
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err,"Failed to get staker client."))
+	}
+	highLevelNormalClient := ava_networks.NewHighLevelGeckoClient(
+		normalClient,
+		stakerUsername,
+		stakerPassword,)
+	err = highLevelNormalClient.GetFundsAndStartValidating(seedAmount, stakeAmount)
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err,"Failed add client as a validator."))
+	}
+	currentStakers, err := normalClient.PChainApi().GetCurrentValidators(nil)
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Could not get current stakers."))
+	}
+	logrus.Debugf("Number of current stakers: %d", len(currentStakers))
+	actualNumStakers := len(currentStakers)
+	expectedNumStakers := 10
+	context.AssertTrue(actualNumStakers == expectedNumStakers, stacktrace.NewError("Actual number of stakers, %v, != expected number of stakers, %v", actualNumStakers, expectedNumStakers))
+}
+func (test StakingNetworkUnrequestedChitSpammerTest) GetNetworkLoader() (testsuite.TestNetworkLoader, error) {
+	return getByzantineNetworkLoader(map[int]int{
+		0: byzantineConfigId,
```

Comment:
you can ever do a for loop in here, with the upper limit (same as above) being `normalNodeServiceId`. No strong feelings though; tests are okay to be a little messy